torch/export/experimental/__init__.py

```diff
@@ -293,18 +295,35 @@ def _exporter_context(*args, **kwargs):  # type: ignore[no-untyped-def]
                     f"Exporter: Cannot export fallback {fn} when fallback policy is set to 'error',"
                     + "please specify an overload or adjust the fallback policy."
                 )
-            elif fallback == "once":
+
+            if fallback == "once":
                 if len(fallbacks) > 0:
                     raise RuntimeError(
                         f"Exporter: Cannot export {fn} more than once, "
                         + "please specify an overload or adjust the fallback policy."
                     )
-            else:
-                raise RuntimeError(f"Unknown fallback policy: {fallback}")
-            ep = torch.export.export(model, args, kwargs)
+                if len(fallbacks) == 0:
+                    ep = torch.export.export(model, args, kwargs)
+                    fallbacks.append(ep)
+                    return ep.module()(*args, **kwargs)
+
+            if fallback == "auto":
+                example_inputs.add(args, kwargs)
+
+                if len(fallbacks) > 0:
+                    ep = fallbacks[0]
+                
+                    if not example_inputs.verify(ep):
+                        ep = torch.export.export(model, args, kwargs, dynamic_shapes=example_inputs)
+                        fallbacks[0] = ep 
```

Comment:
In general overwriting fallbacks seem bad because it's not only shapes that can change across exports. Every time any guard changes, there should be a new fallback produced instead of using the new EP to replace the old EP (which may depend on non-shape invariants).

For example, if I pass 3 inputs in the first run and pass 2 inputs in the second run, we will have very confusing result where we couldn't replay the ExportMethod with the initial 3 inputs anymore. 

At least we should append to the fallback list every time when some metadata on tensor (or the number of tensors) is changed. We should probably never overwrite a fallback list (because non-strict export is unsound and we cannot safely do so anyway)